spyder/plugins/shortcuts/plugin.py

```diff
@@ -191,14 +191,15 @@ def apply_shortcuts(self):
 
             try:
                 if isinstance(qobject, QAction):
-                    if (sys.platform == 'darwin'
-                            and qobject._shown_shortcut == 'missing'):
-                        qobject._shown_shortcut = keyseq
-                    else:
-                        qobject.setShortcut(keyseq)
-
-                    if add_shortcut_to_tip:
-                        add_shortcut_to_tooltip(qobject, context, name)
+                    if qobject.shortcuts() == []:

```

Comment:
Maybe we should add some sort of comment regarding why this changes were needed and also some `TODO` reference comment to improve the way shortcuts get applied in the future?